src/client/components/PageLayout/Header/LanguageSelector/LanguageSelector.tsx

```diff
@@ -1,26 +1,54 @@
-import React from 'react'
+import React, { useCallback, useEffect } from 'react'
 import { useTranslation } from 'react-i18next'
 
+import { AssessmentNames } from 'meta/assessment'
 import { Lang, LanguageCodes } from 'meta/lang'
 
+import { useAssessment } from 'client/store/assessment'
 import Icon from 'client/components/Icon'
 import PopoverControl from 'client/components/PopoverControl'
 
 const LanguageSelector: React.FC = () => {
   const { i18n } = useTranslation()
+  const assessment = useAssessment()
 
-  const updateLanguage = async (lang: Lang) => {
-    await i18n.changeLanguage(lang)
-    await localStorage.setItem('i18n/lang', lang)
-    if (lang === 'ar') document.body.classList.add('rtl')
-    if (lang !== 'ar') document.body.classList.remove('rtl')
-  }
+  const updateLanguage = useCallback(
+    async (lang: Lang) => {
+      await i18n.changeLanguage(lang)
+      await localStorage.setItem('i18n/lang', lang)
+      if (lang === 'ar') document.body.classList.add('rtl')
+      if (lang !== 'ar') document.body.classList.remove('rtl')
+    },
+    [i18n]
+  )
 
   const languageSelectionItems = LanguageCodes.map((lang: Lang) => ({
     content: i18n.t(`language.${lang}`),
     onClick: () => updateLanguage(lang),
   }))
 
+  // If the assessment is panEuropean, we use the english language,
+  // and we don't show the language selector
+  useEffect(() => {
```

Comment:
Can this be done at initialization level ? The language menu should only be hidden and not contain any logic . What do you think ?

Reply:
Introduced utils/updateLanguage
Moved logic to assessment initialisation (highest possible level)